### DISCUSSION TITLE
Sprint Meeting notes 04/05/2022

### DISCUSSION THREAD

**Discussion by VojtechBartoska** [espressif/arduino-esp32#6607] [Sprints Meeting Notes]
Hello Dear Community Users,

If you would like to read fresh news from our development, please check these Sprint Meetings notes :wink:

It covers our progress from last 2 weeks and also shows our plans for next 2 weeks.

## 2.0.3 Evaluation
- Current  tasks progress checked
- No blockers found

## General Releases approach
- Next bug fixes will be released under [2.0.4 milestone](https://github.com/espressif/arduino-esp32/issues?q=is%3Aopen+is%3Aissue+milestone%3A2.0.4)
- Major changes including new features will be released in [2.1.0 milestone](https://github.com/espressif/arduino-esp32/issues?q=is%3Aopen+is%3Aissue+milestone%3A2.1.0)

## Development update
Due to recent release of 2.0.3 stable version, all related issues on the Roadmap were sorted and tasks for next 2 weeks were added into 2.0.4 milestone. Selected issues overview is shown below. 

**As well we will collect feedback on 2.0.3 stable release in upcoming weeks and if any important issue occurs, it will be served with high priority.**

### Tasks Selected for Development
- [Compilation breaks when using -DUSE_ESP_IDF_LOG](https://github.com/espressif/arduino-esp32/issues/4990)
- [Extend String to print 64-bit integers](https://github.com/espressif/arduino-esp32/issues/5672)
- [MDNS special packet from Android App crashes and reboots ESP32/ESP32-S2](https://github.com/espressif/arduino-esp32/issues/6368)
- [Unexpected light sleep and deep sleep behaviour on ESP32-S3](https://github.com/espressif/arduino-esp32/issues/6612)
- [WiFi connection unstable after updating to 2.0.X](https://github.com/espressif/arduino-esp32/issues/6654)
- [DNSServer memory leak](https://github.com/espressif/arduino-esp32/issues/6663)
- [Write a first Test case for Timer and create a manual for HW tests](https://github.com/espressif/arduino-esp32/issues/6686)
- [fixing beginTransaction() thread safety](https://github.com/espressif/arduino-esp32/pull/6425) - Pull Request which needs testing

### Already Under Investigation
- [HTTPClient setTimeout hasn't any effect](https://github.com/espressif/arduino-esp32/issues/1433) - should be fixed, needs retesting
- [Setting DHCP IP-range start for softAP](https://github.com/espressif/arduino-esp32/issues/6271) - merged Pull Request needs to be evaluated
- [analogRead() returns bogus values when i2s DAC running](https://github.com/espressif/arduino-esp32/issues/4782) - testing needed
- [After 1.0.6 update WiFi won't connect to other network // connection time increased](https://github.com/espressif/arduino-esp32/issues/4980) - should be fixed, needs retesting
- [max() not accepting #defined values](https://github.com/espressif/arduino-esp32/issues/6623) - shaping
- [Commit c7cc causes heap corruption](https://github.com/espressif/arduino-esp32/issues/6682) - shaping

### In Progress
NIA

### In Review
- [Added compiler.warning_flags to platform.txt; fixing](https://github.com/espressif/arduino-esp32/pull/6596) - CI issues fixed
- [GetServices crashes in 2.0.0+ when connecting to heart rate monitor](https://github.com/espressif/arduino-esp32/issues/6063)
- [pinMode disables interrupts as a side effect](https://github.com/espressif/arduino-esp32/issues/6669)

**Work on 2.0.4 release is tracked [here](https://github.com/orgs/espressif/projects/3/views/11).**

## General Issue solving progress (longer term)
- [Native USB CDC hangs up on ESP32-S2 after about 320 characters](https://github.com/espressif/arduino-esp32/issues/6221) - shaping (internal tracking)
- [I2C Frequency](https://github.com/espressif/arduino-esp32/issues/6444) - In Review (internal tracking), will be backported to ESP-IDF v4.4
- [Add default 10MHz SD card frequency for SD SPI](https://github.com/espressif/arduino-esp32/issues/6225) - shaping (internal tracking)

## Community and Documentation progress
- More actively solved issues (quick response, assignees, labelling)
- Ongoing Issue tracker cleanup
- First Community Meeting took place (Feedback collected)
- Second Community Meeting is planned on 28/05/2022, any topics can be shared [here](https://github.com/espressif/arduino-esp32/discussions/6675)
- [New docs theme based on ESP-IDF style](https://github.com/espressif/arduino-esp32/pull/6512) - reviewed, old links redirect in progress

### Tasks Selected for Development in Documentation
- [Documentation: tracking issue for filesystem-related libraries API reference](https://github.com/espressif/arduino-esp32/issues/6565)
- [NEED help in setting up partition Scheme/Table for ESP32-WROOM-32D (8MB)](https://github.com/espressif/arduino-esp32/issues/4551)
- [Evalute "Type: For reference" issues and add them to documentation](https://github.com/espressif/arduino-esp32/issues/6538)

Next Sprint Meeting will take place on Wednesday 18/05/2022.

**Comment by mrengineer7777** [comment]
Love seeing what you guys are working on.  Helps us know when issues will be resolved.

**Reply by VojtechBartoska** [reply]
Thanks for feedback and your active and very helpful contribution @mrengineer7777! :)
